cmake/modules/SetupCompiler.cmake

```diff
@@ -105,23 +111,31 @@ endif ()
 # Promote a compiler warning as an error for project targets
 ################################################################
 
-if (DYAD_WARNINGS_AS_ERRORS)
-  dyad_add_cxx_flags(_WERROR_FLAGS -Werror)
-  separate_arguments(_WERROR_FLAGS NATIVE_COMMAND "${_WERROR_FLAGS}")
-  if (NOT TARGET DYAD_CXX_FLAGS_werror)
-    add_library(DYAD_CXX_FLAGS_werror INTERFACE)
-    set_property(TARGET DYAD_CXX_FLAGS_werror PROPERTY
-      INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:${_WERROR_FLAGS}>)
-
-    add_library(DYAD_C_FLAGS_werror INTERFACE)
-    set_property(TARGET DYAD_C_FLAGS_werror PROPERTY
-      INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:C>:${_WERROR_FLAGS}>)
-
-    # Add the "library" to the export
-    install(TARGETS DYAD_C_FLAGS_werror EXPORT ${DYAD_EXPORTED_TARGETS})
-    install(TARGETS DYAD_CXX_FLAGS_werror EXPORT ${DYAD_EXPORTED_TARGETS})
-  endif ()
-endif ()
+macro(dyad_add_werror_if_needed target)
+  if (DYAD_WARNINGS_AS_ERRORS)
+    target_compile_options(${target} PRIVATE 
+      $<$<COMPILER_LANGUAGE:CXX>:"-Werror">
+      $<$<COMPILER_LANGUAGE:C>:"-Werror">)
+  endif()
+endmacro(dyad_add_werror_if_needed target)
+
+# if (DYAD_WARNINGS_AS_ERRORS)
+#   dyad_add_cxx_flags(_WERROR_FLAGS -Werror)
+#   separate_arguments(_WERROR_FLAGS NATIVE_COMMAND "${_WERROR_FLAGS}")
+#   if (NOT TARGET DYAD_CXX_FLAGS_werror)
+#     add_library(DYAD_CXX_FLAGS_werror INTERFACE)
+#     set_property(TARGET DYAD_CXX_FLAGS_werror PROPERTY
+#       INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:${_WERROR_FLAGS}>)
+# 
+#     add_library(DYAD_C_FLAGS_werror INTERFACE)
+#     set_property(TARGET DYAD_C_FLAGS_werror PROPERTY
+#       INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:C>:${_WERROR_FLAGS}>)
+# 
+#     # Add the "library" to the export
+#     install(TARGETS DYAD_C_FLAGS_werror EXPORT ${DYAD_EXPORTED_TARGETS})
+#     install(TARGETS DYAD_CXX_FLAGS_werror EXPORT ${DYAD_EXPORTED_TARGETS})
+#   endif ()
+# endif ()
```

Comment:
Again we can remove this. if not needed.

Reply:
See prior comment

Reply:
Removed